tests/test_xopen.py

```diff
@@ -117,6 +117,14 @@ def test_pipedgzipreader_textiowrapper():
             wrapped = io.TextIOWrapper(f)
             assert wrapped.read() == CONTENT
 
+def test_bytetested_gzip():
```

Comment:
Regarding the added `tests/file.txt....test` files (which I cannot comment on directly): Can you please add them as symbolic links? That would make it clearer that they are actually identical to the versions without the `.test` suffix.